### PR TITLE
 topology2: use passthrough-capture-be instead of passthrough-be 

### PR DESCRIPTION
**Pull request by kv2019i**

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@ranj063 @juimonen pls review.

File: tools/topology/topology2/include/pipelines/cavs/passthrough-capture-be.conf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# }
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+<include/common/audio_format.conf>
+<include/components/copier.conf>
+<include/components/pipeline.conf>
+
+Class.Pipeline."passthrough-capture-be" {
```
Comment (ranj063) [inline tools/topology/topology2/include/pipelines/cavs/passthrough-capture-be.conf:24]: @kv2019i can we call this 32-bit-capture.conf explicitly?

**Comment by kv2019i** [comment]
V2 (Feb 14):
- CI failures (with kernel https://github.com/thesofproject/linux/pull/4137) on SDW and HDA platforms, so extend changes to more topology definitions

File: tools/topology/topology2/include/pipelines/cavs/passthrough-capture-be.conf [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#
+# BE DAI passthrough pipeline for capture
+#
+# All attributes defined herein are namespaced by alsatplg to
+# "Object.Pipeline.passthrough-be-capture.N.attribute_name"
+#
+# Usage: mixin-playback pipeline object can be instantiated as:
```
Comment (kv2019i) [inline tools/topology/topology2/include/pipelines/cavs/passthrough-capture-be.conf:7]: Oops, copy'n'paste error in the doc, will fix in later version.

**Comment by greg-intel** [comment]
SOFCI TEST

File: tools/topology/topology2/include/pipelines/cavs/passthrough-capture-be.conf (hunk 2 above)
Comment (lgirdwood) [reply tools/topology/topology2/include/pipelines/cavs/passthrough-capture-be.conf:7]: Fix incrementally in another PR.

File: tools/topology/topology2/include/pipelines/cavs/passthrough-capture-be.conf (hunk 1 above)
Comment (lgirdwood) [reply tools/topology/topology2/include/pipelines/cavs/passthrough-capture-be.conf:24]: yeah, lets give users a choice if possible. If not, we can hard code this later.

**Review by lgirdwood — APPROVED** [review]

**Review by lyakh — COMMENTED** [review]
do we actually then need passthrough-be?

File: tools/topology/topology2/sof-hda-generic.conf [hunk 3]
```diff
@@ -18,6 +18,7 @@
 <gain-capture.conf>
 <deepbuffer-playback.conf>
 <passthrough-be.conf>
+<passthrough-capture-be.conf>
```
[the remark this answers is not in the document]
Comment (kv2019i) [reply tools/topology/topology2/sof-hda-generic.conf:21]: Yes, you get build error without. Some topologies include pipelines that will use passthrough-be (I think hdmi-generic is one).

**Comment by ranj063** [comment]
> do we actually then need passthrough-be?

Yes, it is used for the playback cases

**Pull request merged**